packages/plugins/Web3Profile/src/SNSAdaptor/utils.ts

```diff
@@ -0,0 +1,310 @@
+                  >
+              >
+          }
+        | undefined,
+    footprints?: Collection[],
+    donations?: Collection[],
+    NFTs?: Collection[],
+) => {
+    if (accounts?.length === 0) return
+    let allLinkedProfiles: ProfileInformation[] = []
+    allPersona?.forEach((persona) => {
+        if (persona?.linkedProfiles) {
+            allLinkedProfiles = [...allLinkedProfiles, ...persona.linkedProfiles]
+        }
+    })
+    const detailedAccounts = accounts?.map((account) => {
+        const linkedProfile = allLinkedProfiles?.find(
+            (profile) =>
+                profile?.identifier?.network?.replace('.com', '') === NextIDPlatform.Twitter &&
+                profile?.identifier?.userId === account?.identity,
+        )
+        return {
+            ...account,
+            linkedProfile,
+        }
+    })
+    return detailedAccounts?.map((key) => ({
+        ...key,
+        walletList: {
+            NFTs: deduplicateArray(wallets, hiddenObj?.hiddenWallets[key?.identity]?.NFTs)?.map((wallet) => ({
+                ...wallet,
+                collections: addHiddenToArray(
+                    NFTs?.find((NFT) => isSameAddress(NFT?.address, wallet?.address))?.collections,
+                    hiddenObj?.hiddenCollections?.[key?.identity]?.[wallet?.address]?.NFTs,
+                ),
+            })),
+            donations: deduplicateArray(wallets, hiddenObj?.hiddenWallets[key?.identity]?.donations)?.map((wallet) => ({
+                ...wallet,
+                collections: addHiddenToArray(
+                    donations?.find((donation) => isSameAddress(donation?.address, wallet?.address))?.collections,
+                    hiddenObj?.hiddenCollections?.[key?.identity]?.[wallet?.address]?.Donations,
+                ),
+            })),
+            footprints: deduplicateArray(wallets, hiddenObj?.hiddenWallets[key?.identity]?.footprints)?.map(
+                (wallet) => ({
+                    ...wallet,
+                    collections: addHiddenToArray(
+                        footprints?.find((footprint) => isSameAddress(footprint?.address, wallet?.address))
+                            ?.collections,
+                        hiddenObj?.hiddenCollections?.[key?.identity]?.[wallet?.address]?.Footprints,
+                    ),
+                }),
+            ),
+        },
+    }))
+}
+
+export const placeFirst = (userId?: string, accountList?: AccountType[]) => {
+    if (!accountList || accountList?.length === 0) return accountList
+    const currentAccountIndex = accountList?.findIndex((account) => account?.identity === userId?.toLowerCase())
+    if (currentAccountIndex === -1) return accountList
+    const currentItem = accountList?.splice(currentAccountIndex, 1)?.[0]
+    return [currentItem, ...accountList]
+}
+
+export const getDonationList = async (walletList: string[]) => {
+    const resNodeIdParams: Collection[] = []
+    const promises = walletList.map((address) => {
```

Comment:
Recommend to use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap.